src/app/stores/store.ts

```diff
@@ -172,6 +172,38 @@ export class Store {
       .find(response => response.uuid === this.store$.value.activeRouteResponseUUID);
   }
 
+  /**
+   * Generate a unused port to a new environment
+   */
+  public getNewEnvironmentPort(): number {
```

Comment:
I would move this in the EnvironmentService class as a private method. Store should stay very basic with no business logic I think.